packages/p2p-v2/src/pages/orders/screens/OrderDetails/OrderDetailsProvider/OrderDetailsProvider.tsx

```diff
@@ -0,0 +1,21 @@
+import React, { createContext, PropsWithChildren, useContext } from 'react';
+import { useExtendedOrderDetails } from '@/hooks';
+
+type TOrderDetails = ReturnType<typeof useExtendedOrderDetails>['data'];
+
+type TContextValue = {
+    isErrorOrderInfo: boolean;
+    orderDetails: TOrderDetails;
+};
+
+const OrderDetailsContext = createContext<TContextValue>({} as TContextValue);
+
+type TOrderDetailsProviderProps = {
+    value: TContextValue;
+};
+
+export const OrderDetailsProvider = ({ children, value }: PropsWithChildren<TOrderDetailsProviderProps>) => (
+    <OrderDetailsContext.Provider value={value}>{children}</OrderDetailsContext.Provider>
+);
+
+export const useOrderDetails = (): TContextValue => useContext(OrderDetailsContext);
```

Comment:
@ameerul-deriv, why not move this file into a folder called `providers` in the `src` directory, seeing it's not a `page`? 🤔  
```suggestion
import React, { createContext, PropsWithChildren, useContext } from 'react';
import { useExtendedOrderDetails } from '@/hooks';

type TOrderDetails = ReturnType<typeof useExtendedOrderDetails>['data'];

type TContextValue = {
    isErrorOrderInfo: boolean;
    orderDetails: TOrderDetails;
};

const OrderDetailsContext = createContext<TContextValue>({} as TContextValue);

type TOrderDetailsProviderProps = {
    value: TContextValue;
};

export const OrderDetailsProvider = ({ children, value }: PropsWithChildren<TOrderDetailsProviderProps>) => (
    <OrderDetailsContext.Provider value={value}>{children}</OrderDetailsContext.Provider>
);

export const useOrderDetails = (): TContextValue => useContext(OrderDetailsContext);
```

Reply:
yeah I think thats true, since now I'm using this in useSendbird as well